Simplenote/SPAppDelegate+Extensions.swift

```diff
@@ -394,3 +394,38 @@ extension SPAppDelegate {
         EditorFactory.shared.scrollPositionCache.cleanup(keeping: allIdentifiers)
     }
 }
+
+// MARK: - Core Data
+//
+extension SPAppDelegate {
+    @objc
+    var managedObjectContext: NSManagedObjectContext {
+        coreDataManager.managedObjectContext
+    }
+
+    @objc
+    func setupStorage() {
+        let migrationResult = SharedStorageMigrator().performMigrationIfNeeded()
+
+        do {
+            try setupCoreData(migrationResult: migrationResult)
+        } catch {
+            fatalError(error.localizedDescription)
+        }
+    }
+
+    private func setupCoreData(migrationResult: MigrationResult) throws {
+        let settings = StorageSettings()
+
+        switch migrationResult {
+        case .notNeeded, .success:
+            do {
+                coreDataManager = try CoreDataManager(settings.sharedStorageURL)
+            } catch {
+                coreDataManager = try CoreDataManager(settings.legacyStorageURL)
+            }
+        case .failed:
+            coreDataManager = try CoreDataManager(settings.legacyStorageURL)
+        }
```

Comment:
I know we changed this already a few times, but having 2 databases doesn't feel right. It's also not a "backup" as in some scenario the legacy becomes the primary one.

So I have another idea. Instead of trying to init a CoreDataManger here and backing off to the legacy storage, we can do that as last step of migration:
* try to init CoreDataManger with the file that we copied
* if it fails, delete new file, return `.failed`
* if it succeeds, delete old file (or move it to a backup folder), return `.success`

And then this code can be simplified to:
```
switch migrationResult {
     case .notNeeded, .success:
          coreDataManager = try CoreDataManager(settings.sharedStorageURL)
     case .failed:
          coreDataManager = try CoreDataManager(settings.legacyStorageURL)
}
```